Common/Thread/Channel.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <mutex>
+#include <condition_variable>
+#include <cassert>
+
+// Named Channel.h because I originally intended to support a multi item channel as
+// well as a simple blocking mailbox. Let's see if we get there.
+
+// Single item mailbox.
+template<class T>
+struct Mailbox {
+	Mailbox() : refcount_(1) {}
+	~Mailbox() {
+		assert(refcount_ == 0);
+	}
+
+	std::mutex mutex_;
+	std::condition_variable condvar_;
+	T *data_ = nullptr;
```

Comment:
I wonder if we could use std::optional and move semantics in some cases instead.  I guess always using a pointer is likely best between threads...

-[Unknown]

Reply:
Yeah, I need to learn how to enforce move semantics in C++ properly. Move-by-default is super-powerful in Rust and getting some of those benefits would be nice...